tianshou/policy/modelfree/bdq.py

```diff
@@ -0,0 +1,184 @@
+from copy import deepcopy
+from typing import Any, Dict, Optional, Union
+
+import numpy as np
+import torch
+
+from tianshou.data import Batch, ReplayBuffer, to_numpy, to_torch_as
+from tianshou.policy import BasePolicy
+
+
+class BDQPolicy(BasePolicy):
```

Comment:
You should definitely inherit from `DQNPolicy` instead of `BasePolicy`. There's tons of duplicate code here which makes everything messy.

Reply:
Yes definitely.
I'll work on this and all the other comments and get back to you. Thanks for the good suggestions!